cli/setup.go

```diff
@@ -851,30 +842,43 @@ func (opts *setupOptionsType) maybeAddHostLookup() {
 	// strip schema and path
 	host := re.ReplaceAllString(opts.serverURL, "$2")
 
+	// Add "s3.SERVER_URL" as well. This is only called in demo mode, so it
+	// should be a safe assumption.
+	route := fmt.Sprintf("%-15s %s s3.%s", opts.serverIP, host, host)
+
+	f, err := os.OpenFile("/etc/hosts", os.O_RDWR, 0644)
+	if err != nil {
+		log.Warnf("Unable to open \"/etc/hosts\" for appending "+
+			"local route \"%s\": %s", route, err.Error())
+		return
+	}
+	defer f.Close()
+	scanner := bufio.NewScanner(f)
+
 	// Check if route allready exists
 	for scanner.Scan() {
 		if strings.Contains(scanner.Text(), host) {
 			return
 		}
 	}
-	route := fmt.Sprintf("\n%-15s %s\n", opts.serverIP, host)
 
 	// Seek to last character
 	_, err = f.Seek(-1, os.SEEK_END)
 	if err != nil {
-		log.Warnf("Unable to add route %q to \"/etc/hosts\": %s",
+		log.Warnf("Unable to add route \"%s\" to \"/etc/hosts\": %s",
```

Comment:
Confused:
Why changing `%q` to explicitly quote a string? 
They're the same thing

Reply:
From the commit message:
```
* Use %s instead of %q. This is actually not critical after the
  newlines were removed, but is still semantically better: Quoted
  strings are definitely not appropriate to copy-paste into
  `/etc/hosts`.
```

Reply:
haha, thx @lluiscampos. However, I still don't get it. The output is the exact same

Reply:
Actually the `%q` has nothing to do with the quotes, these were missing in both bases. The `%q` has to do with newlines. With newlines, in the `%s` case, you get:

```
Unable to add route 
127.0.0.1 docker.mender.io
 to \"/etc/hosts\"
```

With `%q`, you get:

```
Unable to add route \n127.0.0.1 docker.mender.io\n to \"/etc/hosts\"
```

In the end I decided to remove the newlines from the output, which is why this is not critical to fix. However, from a general perspective it's clear that `%s` gives more correct output for this particular case, since `\n` will be misinterpreted inside `/etc/hosts`. If it were a C-string, then `%q` would have been more appropriate.